### DISCUSSION TITLE
How to reduce build output rootfs.wic image ?

### DISCUSSION THREAD

**Discussion by Micko8** [polarfire-soc/polarfire-soc-documentation#231] [Linux build systems for PolarFire SoC] [answered]
Hello all,

I managed to build a RiscV linux image with Yocto Bitbake (MACHINE=m100pfsevp bitbake mpfs-dev-cli) but this image is 5GB of size.

My colleague can have one image less than 1 GB of size (unzipped of course). According to him, the "tip" is to configure the local.conf this way :
_TMPDIR = "/home/micko8/aries-yocto-outputs/tmp"_  Same path for SSTATE_DIR and DL_DIR, basically out of the build (TOPDIR) directory.

When I do the same, I keep getting a 5GB image.

I have also tried by configuring  local.conf.extented this way: 
_MKLIBS_OPTIMIZED_IMAGES ?= "mpfs-dev-cli"_   (it's specific to my board's bsp)

Do you guys have a tip to reduce image size or even the **tmp-glibc** itself ? 

Also if someone could explain me what Yocto is doing with all the recipes involved to ends up with a such size ? Does it duplicate them ?
I need to find a way to optimize this because my machine does not have the space required to install any other app or software.

Thank you for your help !

#Yoctobeginner

**Comment by lhanlyu** [comment, marked as answer]
DL-DIR (Download Directory) and State cache is noting to do with the finished image size. 
You can try bitbake 'core-image-minimal-dev' and the image size is smaller or other bitbake images 'core-image-minimal' it all depends in what you want included in the build.

**Comment by Micko8** [comment]
Thank you @lhanlyu 
I will try to find other parameters to use in local.conf.
